pkg/minikube/registry/drvs/hyperv/vswitch.go

```diff
@@ -0,0 +1,212 @@
+// +build windows
+
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hyperv
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type netAdapter struct {
+	InterfaceGUID        string `json:"interfaceGuid"`
+	InterfaceDescription string
+}
+
+type vmSwitch struct {
+	Name                    string
+	NetAdapterInterfaceGuid []string
+}
+
+func getNetAdapters(physical bool, condition string) ([]netAdapter, error) {
+	//	Get-NetAdapter -Physical | Where-Object {($_.Status -eq "Up") -And  ($_.PhysicalMediaType -notlike "*802.11*")} | Select-Object -Property InterfaceDescription
+	//	ConvertTo-Json @(Hyper-V\Get-VMSwitch | Where-Object { $_.SwitchType -eq 2 -And $_.NetAdapterInterfaceDescriptions -contains $Adapters[0].InterfaceDescription })
+	cmdlet := []string{"Get-NetAdapter"}
+	if physical {
+		cmdlet = append(cmdlet, "-Physical")
+	}
+	cmd := []string{strings.Join(cmdlet, " ")}
+	if condition != "" {
+		cmd = append(cmd, fmt.Sprintf("Where-Object {%s}", condition))
+	}
+	cmd = append(cmd, "Select-Object -Property InterfaceGuid, InterfaceDescription")
+	stdout, err := cmdOut(fmt.Sprintf("ConvertTo-Json @(%s)", strings.Join(cmd, " | ")))
+	if err != nil {
+		return nil, err
+	}
+
+	var adapters []netAdapter
+	err = json.Unmarshal([]byte(strings.TrimSpace(stdout)), &adapters)
+	if err != nil {
+		return nil, err
+	}
+
+	return adapters, nil
+}
+
+func getVMSwitch(condition string) ([]vmSwitch, error) {
+	cmd := []string{"Hyper-V\\Get-VMSwitch"}
+	if condition != "" {
+		cmd = append(cmd, fmt.Sprintf("Where-Object {%s}", condition))
+	}
+	cmd = append(cmd, "Select-Object -Property Name, NetAdapterInterfaceGuid")
+	stdout, err := cmdOut(fmt.Sprintf("ConvertTo-Json @(%s)", strings.Join(cmd, " | ")))
+	if err != nil {
+		return nil, err
+	}
+
+	var vmSwitches []vmSwitch
+	err = json.Unmarshal([]byte(strings.TrimSpace(stdout)), &vmSwitches)
+	if err != nil {
+		return nil, err
+	}
+
+	return vmSwitches, nil
+}
+
+func findConnectedVMSwitch(adapterGUID string) (string, error) {
+	foundSwitches, err := getVMSwitch(fmt.Sprintf("($_.SwitchType -eq 2) -And ($_.NetAdapterInterfaceGuid -contains \"%s\")", adapterGUID))
+	if err != nil {
+		return "", err
+	}
+
+	if len(foundSwitches) > 0 {
+		return foundSwitches[0].Name, nil
+	}
+
+	return "", nil
+}
+
+func getOrderedAdapters() ([]netAdapter, error) {
```

Comment:
Can you add a docstring for each of these functions?

I'm unsure what the difference between `getOrderedAdapters` and `getNetAdapters` is.

Reply:
Done